src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -1361,21 +1360,27 @@ abstract class BCodeBodyBuilder extends BCodeSkelBuilder {
     def genLoadTry(tree: Try): BType
 
     def genInvokeDynamicLambda(canLMF: delambdafy.LambdaMetaFactoryCapable) = {
-      import canLMF._
+      import canLMF.{ lambdaTarget => originalTarget, _ }
 
-      val isStaticMethod = lambdaTarget.hasFlag(Flags.STATIC)
+      val lambdaTarget = originalTarget.attachments.get[JustMethodReference].map(_.lambdaTarget).getOrElse(originalTarget)
       def asmType(sym: Symbol) = classBTypeFromSymbol(sym).toASMType
 
       val isInterface = lambdaTarget.owner.isTrait
+      val tag =
+        if (lambdaTarget.isStaticMember) Opcodes.H_INVOKESTATIC
+        else if (lambdaTarget.isPrivate) Opcodes.H_INVOKESPECIAL
+        //else if (lambdaTarget.isClassConstructor) Opcodes.H_NEWINVOKESPECIAL // to invoke Foo::new directly
```

Comment:
@dwijnand Dale, was there a problem with this?

Reply:
It didn't seem to have any effect on an example I made, i.e. it just emitting old style, so I just left a breadcrumb.

Reply:
Well

```scala
case Apply(fun @ Select(New(tpt), nme.CONSTRUCTOR), args) =>
```

is matched before

```scala
case Apply(fun, args) if app.hasAttachment[delambdafy.LambdaMetaFactoryCapable] =>
```

So probably that's the first change to try.